sys-libs/mesa/patches/mesa-19.1.7.patchset

```diff
@@ -0,0 +1,414 @@
+ 
+-	struct st_context *stContext = (struct st_context*)context->st;
+-
+-	unsigned nColorBuffers = stContext->state.framebuffer.nr_cbufs;
+-	for (unsigned i = 0; i < nColorBuffers; i++) {
+-		pipe_surface* surface = stContext->state.framebuffer.cbufs[i];
+-		if (!surface) {
+-			ERROR("%s: Color buffer %d invalid!\n", __func__, i);
+-			continue;
+-		}
+-
+-		TRACE("%s: Flushing color buffer #%d\n", __func__, i);
+-
+-		// We pass our destination bitmap to flush_fronbuffer which passes it
+-		// to the private winsys display call.
+-		fScreen->flush_frontbuffer(fScreen, surface->texture, 0, 0,
+-			context->bitmap, NULL);
++	struct hgl_buffer* buffer = hgl_st_framebuffer(context->draw->stfbi);
++	pipe_surface* surface = buffer->surface;
++	if (!surface) {
++		ERROR("%s: Invalid drawable surface!\n", __func__);
++		return B_ERROR;
+ 	}
+ 
+-	#if 0
+-	// TODO... should we flush the z stencil buffer?
+-	pipe_surface* zSurface = stContext->state.framebuffer.zsbuf;
+-	fScreen->flush_frontbuffer(fScreen, zSurface->texture, 0, 0,
++	fScreen->flush_frontbuffer(fScreen, surface->texture, 0, 0,
+ 		context->bitmap, NULL);
+-	#endif
+ 
+ 	return B_OK;
+ }
+diff --git a/src/gallium/state_trackers/hgl/hgl_context.h.older b/src/gallium/state_trackers/hgl/hgl_context.h
+index 5acb2f0..c5995f1 100644
+--- a/src/gallium/state_trackers/hgl/hgl_context.h.older
++++ b/src/gallium/state_trackers/hgl/hgl_context.h
+@@ -39,6 +39,8 @@ struct hgl_buffer
+ 	unsigned mask;
+ 
+ 	struct pipe_screen* screen;
++	struct pipe_surface* surface;
++
+ 	enum pipe_texture_target target;
+ 	struct pipe_resource* textures[ST_ATTACHMENT_COUNT];
+ 
+@@ -80,6 +82,8 @@ struct hgl_context
+ 	struct hgl_buffer* read;
+ };
+ 
++// hgl_buffer from statetracker interface
++struct hgl_buffer* hgl_st_framebuffer(struct st_framebuffer_iface *stfbi);
+ 
+ // hgl state_tracker api
+ struct st_api* hgl_create_st_api(void);
+diff --git a/src/mapi/glapi/glapi_dispatch.c.older b/src/mapi/glapi/glapi_dispatch.c
+index ca15449..2489524 100644
+--- a/src/mapi/glapi/glapi_dispatch.c.older
++++ b/src/mapi/glapi/glapi_dispatch.c
+@@ -153,6 +153,9 @@ GL_API void GL_APIENTRY glPointSizePointerOES (GLenum type, GLsizei stride, cons
+ GL_API void GL_APIENTRY glBlendBarrier (void);
+ GL_API void GL_APIENTRY glPrimitiveBoundingBox (GLfloat minX, GLfloat minY, GLfloat minZ, GLfloat minW, GLfloat maxX, GLfloat maxY, GLfloat maxZ, GLfloat maxW);
+ 
++GL_API void GL_APIENTRY glBlendBarrier (void);
++GL_API void GL_APIENTRY glPrimitiveBoundingBox (GLfloat minX, GLfloat minY, GLfloat minZ, GLfloat minW, GLfloat maxX, GLfloat maxY, GLfloat maxZ, GLfloat maxW);
++
+ /* Enable frame pointer elimination on Windows, otherwise forgetting to add
+  * APIENTRY to _mesa_* entrypoints will not cause crashes on debug builds, as
+  * the initial ESP value is saved in the EBP in the function prologue, then
+
+diff --git a/src/gallium/state_trackers/hgl/Makefile.am b/src/gallium/state_trackers/hgl/Makefile.am
```

Comment:
Umm, MESA switched from auto* to Meson, all the Makefiles should be rewritten for meson.

Reply:
Fixed.